src/en/anchira/src/eu/kanade/tachiyomi/extension/en/anchira/Anchira.kt

```diff
@@ -0,0 +1,342 @@
+import eu.kanade.tachiyomi.extension.en.anchira.AnchiraHelper.json
+import eu.kanade.tachiyomi.extension.en.anchira.AnchiraHelper.prepareTags
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.network.interceptor.rateLimit
+import eu.kanade.tachiyomi.source.ConfigurableSource
+import eu.kanade.tachiyomi.source.model.Filter
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.MangasPage
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.model.UpdateStrategy
+import eu.kanade.tachiyomi.source.online.HttpSource
+import kotlinx.serialization.json.decodeFromStream
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.Interceptor
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.Response
+import uy.kohesive.injekt.Injekt
+import uy.kohesive.injekt.api.get
+import java.io.IOException
+import java.util.concurrent.TimeUnit
+
+class Anchira : HttpSource(), ConfigurableSource {
+    override val name = "Anchira"
+
+    override val baseUrl = "https://anchira.to"
+
+    private val apiUrl = "$baseUrl/api/v1"
+
+    private val libraryUrl = "$apiUrl/library"
+
+    private val cdnUrl = "https://kisakisexo.xyz"
+
+    override val lang = "en"
+
+    override val supportsLatest = true
+
+    override val client: OkHttpClient = network.cloudflareClient.newBuilder()
+        .rateLimit(3, 1, TimeUnit.SECONDS)
+        .addInterceptor { apiInterceptor(it) }
+        .build()
+
+    private val preferences: SharedPreferences by lazy {
+        Injekt.get<Application>().getSharedPreferences("source_$id", 0x0000)
+    }
+
+    override fun headersBuilder() = super.headersBuilder().add("X-Requested-With", "XMLHttpRequest")
+
+    // Latest
+
+    override fun latestUpdatesRequest(page: Int) = GET("$libraryUrl?page=$page", headers)
+
+    override fun latestUpdatesParse(response: Response): MangasPage {
+        val data = decodeBytes<LibraryResponse>(response.body)
+
+        return if (data.entries != null) {
+            MangasPage(
+                data.entries.map {
+                    SManga.create().apply {
+                        url = "/g/${it.id}/${it.key}"
+                        title = it.title
+                        thumbnail_url = "$cdnUrl/${it.id}/${it.key}/m/${it.cover.name}"
+                        artist = it.tags.filter { it.namespace == 1 }.joinToString(", ") { it.name }
+                        author = it.tags.filter { it.namespace == 2 }.joinToString(", ") { it.name }
+                        genre = prepareTags(it.tags)
+                        update_strategy = UpdateStrategy.ONLY_FETCH_ONCE
+                        status = SManga.COMPLETED
+                        initialized = false
+                    }
+                }.toList(),
+                data.entries.size < data.total,
```

Comment:
Would this not always return `true`, because the size of each page is necessarily always smaller than the total number of galleries?

If there is a fixed size for each page (say 25), I would recommend something like this instead:
```kotlin
(page + 1) * 25 < data.total
```